pkg/proxy/iptables/proxier.go

```diff
@@ -962,6 +986,16 @@ func (proxier *Proxier) deleteEndpointConnections(connectionMap map[endpointServ
 	}
 }
 
+// hostAddress returns a host address of the form <ip-address>/32 for
+// IPv4 and <ip-address>/128 for IPv6
+func hostAddress(ip net.IP) string {
+	len := 32
+	if ip.To4() == nil {
```

Comment:
Is it always guaranteed that ip is a valid IP address? IOW no need to check for nil? Just wondering if all the callers using this are providing valid IP addresses.

Reply:
The nil check isn't being done as a sanity check, it's being done to determine if the IP address is IPv4 or IPv6.